aikau/src/main/resources/alfresco/documentlibrary/_AlfDndDocumentUploadMixin.js

```diff
@@ -579,16 +581,144 @@ define(["dojo/_base/declare",
+            {
+               // Top of drop target is below the top of scroll area
+               top = dndNodeBoundingRect.top;
+               if (dndNodeBoundingRect.bottom >= scrollAreaBoundingRect.bottom)
+               {
+                  // ...the bottom of the drop target is BELOW that of the scroll area... this means
+                  // that we just need to place the overlay over the ENTIRE scroll area...
+                  height = scrollAreaBoundingRect.bottom - dndNodeBoundingRect.top;
+               }
+               else
+               {
+                  // ...the bottom of the drop target is ABOVE that of the scroll area, this means
+                  // that we shouldn't overlay all the way to the bottom of the scroll area...
+                  height = dndNodeBoundingRect.bottom - dndNodeBoundingRect.top;
+               }
+            }
+            else
+            {
+               // Top of drop target is above the top of the scroll area (this implies that it has been scrolled)...
+               top = scrollAreaBoundingRect.top;
+
+               if (dndNodeBoundingRect.bottom >= scrollAreaBoundingRect.bottom)
+               {
+                  // ...the bottom of the drop target is BELOW that of the scroll area... this means
+                  // that we just need to place the overlay over the ENTIRE scroll area...
+                  height = scrollAreaBoundingRect.bottom - scrollAreaBoundingRect.top;
+               }
+               else
+               {
+                  // ...the bottom of the drop target is ABOVE that of the scroll area, this means
+                  // that we shouldn't overlay all the way to the bottom of the scroll area...
+                  height = dndNodeBoundingRect.bottom - scrollAreaBoundingRect.top;
+               }
+            }
+            
+         }
+
          domStyle.set(this.dragAndDropOverlayNode, {
-            height: dndNodeDimensions.h + "px",
+            height: height + "px",
             width: dndNodeDimensions.w + "px",
-            top: dndNodePosition.y + "px",
+            top: top + "px",
             left: dndNodePosition.x + "px"
          });
       },
+
+      /**
+       * This function recursively searches out through the DOM to find the first parent of the supplied element that
+       * is capable of scrolling vertically and has scrollbars displayed.
+       * 
+       * @instance
+       * @returns {element} The DOM element that is the scroll parent with scroll bars displayed
+       * @since 1.0.60
+       */
+      findScrollParent: function alfresco_documentlibrary__AlfDndDocumentUploadMixin__findScrollParent(domNode) {
+         var scrollParent = $(domNode).scrollParent();
+         if (!scrollParent.is("html") &&
+             scrollParent[0].clientHeight === scrollParent[0].scrollHeight)
+         {
+            scrollParent = this.findScrollParent(scrollParent[0]);
+         }
+         return scrollParent;
+      },
```

Comment:
This doesn't check that the element can scroll, only that it's not displaying its entire content. Have a look at the `hasScrollbars()` method in `TestCommon` to see a more definitive test. I also worry that a scrollHeight/contentHeight equality test might occasionally fail because the content is too small for a container? Not 100% sure about that though.